internal/circle/client.go

```diff
@@ -177,22 +182,85 @@ func (client *Client) GetWalletByID(ctx context.Context, id string) (*Wallet, er
 	return parseWalletResponse(resp)
 }
 
+type RetryableError struct {
+	err        string
+	retryAfter time.Duration
+}
+
+func (err RetryableError) Error() string {
+	return err.err
```

Comment:
```suggestion
	return fmt.Errorf("retriable error: %w", err.err)
```

Reply:
Addressed in 2bfdc05b0155ff373211ec3154658d97715633ae